mono/metadata/threadpool-io.h

```diff
@@ -15,6 +15,10 @@ typedef struct _MonoIOSelectorJob MonoIOSelectorJob;
 
 TYPED_HANDLE_DECL (MonoIOSelectorJob);
 
+ICALL_EXPORT
+void
+ves_icall_System_IOSelector_Add (gpointer handle, MonoIOSelectorJobHandle job_handle, MonoError* error);
```

Comment:
Icalls should be defined in the icall-def files. This particular function appears to already be defined there - do the signatures not match? I don't see any error when building on OSX at least.

Reply:
They do match, but apparently threadpool-io.c does not include icall-decl.h header like, I'll fix it up. I get this warning with gcc 9.2.1, that's probably not what you're using on osx. To clean it up one step further, should I remove ves_icall_System_IOSelector_Remove() aswell, moving it to icall-decl.h?

Reply:
No. remove is different, as I recall -- NOHANDLES vs. HANDLES.

Reply:
I think you can move the NOHANDLES ones as well? Not sure what the convention is though.
Looking further, the warning also appears to show up on netcore, but in that case I think the implementation needs to be ifdef'd out? Not sure what IOSelector is, but it looks like some mono-specific legacy thing.
```
threadpool-io.c:603:1: warning: no previous prototype for function 'ves_icall_System_IOSelector_Add' [-Wmissing-prototypes]
ves_icall_System_IOSelector_Add (gpointer handle, MonoIOSelectorJobHandle job_handle, MonoError* error)
```

Reply:
Just to make sure I understand, do you want me to move _Remove() to icall-decl.h? (out of threadpool-io.h is the point, I think)